### PR TITLE
Issue 44540: More options for viewing the pipeline log

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
Very large pipeline job logs cause problems for the detail page's log viewer

#### Changes
* Offer direct link to raw log file
* Centralize WebDav URL generation
* Cap detail page's log viewer at 10MB

### CODE REVIEW

**Comment by labkey-jeckels** [comment]
@bbimber here's the updated PR

**Comment by bbimber** [comment]
thanks @labkey-jeckels

File: pipeline/src/org/labkey/pipeline/status/StatusDetailsBean.java [hunk 1]
```diff
@@ -164,20 +186,35 @@ public static StatusDetailsBean create(Container c, PipelineStatusFile psf, long
...
 
-            return br.transferTo(pw);
+            try
+            {
+                return br.transferTo(pw);
+            }
+            catch (LimitedSizeInputStream.LimitReachedException e)
+            {
+                String extraMessage = System.lineSeparator() + System.lineSeparator() +
+                        "Log file viewing is capped at " + MAX_LOG_SIZE + " bytes. Use the raw the full file to view the remainder locally."
```
Comment (labkey-adam) [inline pipeline/src/org/labkey/pipeline/status/StatusDetailsBean.java]: Garbled comment

**Review by labkey-adam — APPROVED** [review]
Please adjust the garbled message, but no further review needed

File: pipeline/src/org/labkey/pipeline/status/StatusDetailsBean.java (hunk 1 above)
Comment (labkey-jeckels) [reply pipeline/src/org/labkey/pipeline/status/StatusDetailsBean.java]: Fixed.

**Pull request merged**